src/common/com/intellij/plugins/haxe/model/HaxePackageModel.java

```diff
@@ -125,9 +125,23 @@ public HaxeFileModel getFileModel(String fileName) {
   protected HaxeFile getFile(String fileName) {
     PsiDirectory directory = root.access(path);
     if (directory != null && directory.isValid()) {
-      PsiFile file = directory.findFile(fileName + ".hx");
-      if (file != null && file.isValid() && file instanceof HaxeFile) {
-        return (HaxeFile)file;
+      PsiFile file;
+      try {
+        String fName = fileName + ".hx";
+        //TODO @ Eric.  What could be the source of a request for "Unknown.hx" ?
+        if(fName.equals("Unknown.hx")){
+          return null;
+        }
+        //file = directory.findFile(fileName + ".hx");
+        file = directory.findFile(fName);
+        if (file != null && file.isValid() && file instanceof HaxeFile) {
+          return (HaxeFile)file;
+        }
+      } catch(Exception e){
+        //System.out.println("-----------------------------------");
+        //System.out.println(fileName + ".hx");
+        //System.out.println("-----------------------------------");
+        System.out.println(Arrays.toString(e.getStackTrace()));
```

Comment:
Always use the HaxeDebugLogger instead of any of the System io dumps (out, err); the System calls can create write dead-locks when multiple JetBrains products are used simultaneously or when called on background threads (for example, see #724).  They are also not captured in user logs where they could be useful for debugging.

`HaxeDebugLogger.getLogger().warn("Message");` is the easy way to add a temporary message.  There are many examples of more interesting usage throughout the code base.

Reply:
Cool..thanks for that tip.